aea/aea_builder.py

```diff
@@ -0,0 +1,746 @@
+        self._prefix_to_components.get(component_id.component_prefix, set()).discard(
+            component_id
+        )
+        # update inverse dependency graph
+        for dependency in component.configuration.package_dependencies:
+            self._inverse_dependency_graph[dependency].discard(component_id)
+
+    def check_package_dependencies(
+        self, component_configuration: ComponentConfiguration
+    ) -> bool:
+        """
+        Check that we have all the dependencies needed to the package.
+
+        return: True if all the dependencies are covered, False otherwise.
+        """
+        not_supported_packages = component_configuration.package_dependencies.difference(
+            self.all_dependencies
+        )
+        return len(not_supported_packages) == 0
+
+    @property
+    def pypi_dependencies(self) -> Dependencies:
+        """Get all the PyPI dependencies."""
+        all_pypi_dependencies = {}  # type: Dependencies
+        for component in self._dependencies.values():
+            # TODO implement merging of two PyPI dependencies.
+            all_pypi_dependencies.update(component.configuration.pypi_dependencies)
+        return all_pypi_dependencies
+
+    @contextmanager
+    def load_dependencies(self):
+        """
+        Load dependencies of a component, so its modules can be loaded.
+
+        :return: None
+        """
+        modules = self._get_import_order()
+        with _SysModules.load_modules(modules):
+            yield
+
+    def _get_import_order(self) -> List[Tuple[str, types.ModuleType]]:
+        """
+        Get the import order.
+
+        At the moment:
+        - protocols and contracts don't have dependencies.
+        - a connection can depend on protocols.
+        - a skill can depend on protocols and contracts.
+
+        :return: a list of pairs: (import path, module object)
+        """
+        # get protocols first
+        components = filter(
+            lambda x: x in self.dependencies_highest_version,
+            itertools.chain(
+                self.protocols.values(),
+                self.contracts.values(),
+                self.connections.values(),
+                self.skills.values(),
+            ),
+        )
+        module_by_import_path = [
+            (self._build_dotted_part(component, relative_import_path), module_obj)
+            for component in components
+            for (
+                relative_import_path,
+                module_obj,
+            ) in component.importpath_to_module.items()
+        ]
+        return cast(List[Tuple[str, types.ModuleType]], module_by_import_path)
+
+    @staticmethod
+    def _build_dotted_part(component, relative_import_path) -> str:
+        """Given a component, build a dotted path for import."""
+        if relative_import_path == "":
+            return component.prefix_import_path
+        else:
+            return component.prefix_import_path + "." + relative_import_path
```

Comment:
in fact, this will only be used when there are Python sub-packages in a package directory.